Add plugin: TubeSage

rmccorkl opened the pull request:
TubeSage create comprehensive notes from YouTube transcripts them using LLMs.

# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/rmccorkl/TubeSage
## Release Checklist
- [ x] I have tested the plugin on
  - [ x]  Windows
  - [ x]  macOS
  - [ x]  Linux
  - [ ]  Android _(if applicable)_
  - [ x]  iOS _(if applicable)_
- [ x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x ] `main.js`
  - [x ] `manifest.json`
  - [x ] `styles.css` _(optional)_
- [x ] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [ x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x ] My README.md describes the plugin's purpose and provides clear usage instructions.
- [ x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [ x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [ x] I have added a license in the LICENSE file.
- [ x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

rmccorkl commented:
Bump Release Update version to 1.0.1 in manifest.json and package.json for release preparation.

rmccorkl commented:
chore: update version to 1.0.4 in manifest.json and package.json; add italicized mission statement to settings UI and new CSS class for styling. Added Contribution, Issue docs and updates

joethei commented:
I don't see 

[return manifestData.version || 'Unknown';](https://github.com/rmccorkl/TubeSage/blob/70f746b7d827891a7ac9fa815c64b3055923fd84/main.ts#L262)
Use `this.manifest.version` to get the version number.

[const styleEl = document.createElement('style');](https://github.com/rmccorkl/TubeSage/blob/70f746b7d827891a7ac9fa815c64b3055923fd84/main.ts#L290-L761)
This should be part of the `styles.css` file

[// Utility method to detect if we're running on a mobile device](https://github.com/rmccorkl/TubeSage/blob/70f746b7d827891a7ac9fa815c64b3055923fd84/main.ts#L2502-L2513)
Use the Platform API and remove the fallback, it's not necessary.
And remove the `as any` cast.

[this.addRibbonIcon('youtube', 'TubeSage: Youtube Note Creator', () => {](https://github.com/rmccorkl/TubeSage/blob/70f746b7d827891a7ac9fa815c64b3055923fd84/main.ts#L764), [contentEl.createEl('h2', { text: 'TubeSage: Create Note from YouTube Transcript' });](https://github.com/rmccorkl/TubeSage/blob/70f746b7d827891a7ac9fa815c64b3055923fd84/main.ts#L2688), ['Extensive Summary Prompts'](https://github.com/rmccorkl/TubeSage/blob/70f746b7d827891a7ac9fa815c64b3055923fd84/main.ts#L4465), ['Fast Summary Prompts'](https://github.com/rmccorkl/TubeSage/blob/70f746b7d827891a7ac9fa815c64b3055923fd84/main.ts#L4404)
[Use sentence case in UI](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Use+sentence+case+in+UI)

[// Add version display under the title](https://github.com/rmccorkl/TubeSage/blob/70f746b7d827891a7ac9fa815c64b3055923fd84/main.ts#L3632-L3694)
This should not be part of the settings tab.

joethei commented:
[ const styleEl = document.createElement('style');](https://github.com/rmccorkl/TubeSage/blob/740efd85c4a69fdd32bed2c47d3a0ab6874f1470/main.ts#L290-L293)
This is not necessary, Obsidian will already load that file.

joethei commented:
Sorry, but that's just not true, something else is breaking the CSS.
The bug you are linking to is part of another plugin, not the Obsidian app, which was fixed 4 years ago...
And that header should not be there anyway.

rmccorkl commented:
Okay but can I keep my implementation is it that big a problem from your perspective?

joethei commented:
Remove the settings tab header and the manual CSS loading.

rmccorkl commented:
### **Plugin CSS Load Issues in Obsidian**

I have carefully tested and analyzed the behavior of CSS applied through my Obsidian plugin’s styles.css. Despite following best practices and refactoring everything for high-specificity selectors, my styles are either never applied or are immediately overridden.

**1. What I Did**

Refactored the  plugin CSS using very specific selectors, below is example used on one element

```
.tubesage-settings-container h1.tubesage-settings-main-title {
    font-size: 32px !important;
    font-weight: 700 !important;
    margin: 20px auto 30px auto !important;
    color: var(--text-normal) !important;
    text-align: center !important;
    display: block !important;
    width: 100% !important;
}
```
I've even tried scoping to a know context which also fails e.g

`.workspace .mod-settings .tubesage-settings-container h1.tubesage-settings-main-title { ... }
`

These selectors use container classes, element names, and !important flags across all properties. This should provide ultra-high specificity and override most competing rules.

**2, What I Observed**
	•	In DevTools, my CSS rules do not appear in the list, and so not applied.
	•	Even after switching to container-based scoping and calling out the element (h1), the styles do not persist.
	•	The issue seems to be not specificity, but load order: so _possibly_ my plugin CSS is loaded before theme and snippet CSS, and so is overwritten.

**3. What Actually Works to date**
	a) Inline styles 
        b) Manually injected runtime elements

```
const styleEl = document.createElement("style");
styleEl.textContent = `h1.tubesage-settings-main-title { text-align:center; }`;
document.head.append(styleEl);
```

That reliably apply my intended styles. Both of these approaches load after theme/snippet CSS, so they win the cascade. However, both are against best practices.

Given the findings can you advise on next steps I agree with best practice but its clearly from my limited understanding of CSS the automatic loading is not working. Thank you for your consideration

Automatic Loading shows that app.css which is the obsidian internal CSS
![Screenshot 2025-06-23 at 12 45 26](https://github.com/user-attachments/assets/a7284258-55fb-412a-94b1-d13ba69c1d35)

Manual Loa
![Screenshot 2025-06-23 at 15 33 57](https://github.com/user-attachments/assets/88ccbe17-28f0-4020-be23-9074a4a13413)
ding (when a revert the code to manual load you can see the override of style over  h1 on app.css)

rmccorkl commented:
Removed Title of Plugin from settings

rmccorkl commented:
I’ve published the tagged release (1.0.8) on my repo: https://github.com/rmccorkl/TubeSage/releases/tag/1.0.8
Please let me know if anything else is needed!

rmccorkl commented:
@joethei All requested changes are in. Thanks for another look.

rmccorkl commented:
Updated release 1.0.9

rmccorkl commented:
Refactor YouTubeTranscript plugin settings UI: streamline info icon integration, enhance tooltip handling with CSS, and update modal styling for improved mobile responsiveness. Bump version to 1.1.0 in manifest.json and package.json.

rmccorkl commented:
Bump release 1.1.2

rmccorkl commented:
Bump Release 1.1.3 release of Tubesage updated timestamp prompt

joethei commented:
[// Add CSS for the modal](https://github.com/rmccorkl/TubeSage/blob/4cfcdc6f32ded9e7f8df161b5000d62d66b7548d/main.ts#L321-L325)
This is not necessary, Obsidian already handles that

rmccorkl commented:
Updated description fixed code

rmccorkl commented:
Sync description fixed code

rmccorkl commented:
fixed description manifest

rmccorkl commented:
Bump release of TubeSage 1.1.8 shorts/live/custom URLs that match returned null

rmccorkl commented:
Bump release of TubeSage 1.1.9 more obsidian bot constraints"

rmccorkl commented:
Bump release of TubeSage 1.2.0

rmccorkl commented:
Bump release of TubeSage 1.2.0 more obsidian bot constraints

rmccorkl commented:
Bump release to address bots findings

rmccorkl commented:
Bump release to address obsidian bots findings it does not allow acronyms e.g LLM

rmccorkl commented:
Code health pass

Removed 25 unused exports across src/utils/ and src/llm/
Removed 11 unused npm dependencies (@langchain/anthropic, @langchain/google-genai, langchain, openai, node-html-parser, parse-duration, @anthropic-ai/tokenizer, @typescript-eslint/eslint-plugin, @typescript-eslint/parser, adm-zip)
Cleaned up dead function bodies; lint now passes 0/0 on cleanup work
UI text fixes for Obsidian community plugin review

YouTube → Youtube in setting names
Scrapecreators → Scrape creators
Supadata → Supa data
OpenAI → Openai in provider dropdown
OAUTH → oauth
No functional changes. Plugin behavior is identical to 1.2.24.

rmccorkl commented:
What changed
UI text fix for ObsidianReviewBot

The bot has been rejecting uppercase API acronym casing in UI strings, so lowercased to api on the four bot-flagged setting names and description text:

Youtube data API key → Youtube data api key
Scrape creators API key → Scrape creators api key
Supa data API key → Supa data api key
Description text: Google cloud console API key → Google cloud console api key
No functional changes — UI text only.

rmccorkl commented:
UI text fix for ObsidianReviewBot — restore uppercase API

rmccorkl commented:
/skip                                                                                                                                                               
   
 ESLint and the bot are in contradiction on these two items.                                                                                                         
                                                                                           
 Line 4072 — 'Youtube data API key' (and the matching description on 4073)                                                                                           
  The local obsidianmd/ui/sentence-case rule auto-fixes this to uppercase API. A previous bot scan flagged API as a sentence-case violation; I changed it to api and ESLint then rejected it. Whichever casing I commit, one of the two checks rejects it.                                                                               
   
 Line 4222 — dropdown.addOption('openai', 'Openai')                                                                                                                  
 The local rule auto-fixes the label to 'OpenAI' (canonical brand). A previous bot scan flagged 'OpenAI' as a sentence-case violation, so I reverted to 'Openai'. ESLint now wants it back to 'OpenAI'.                                                                                                                               
                                                                                           
Currently shipping API (per linter) and 'Openai' (per previous bot complaint). Requesting human review since the automated checks give opposite verdicts on the same strings.
